From e6633f03b2537e10956b38e9b2d5f2ec6809bb7b Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@collabora.com>
Date: Fri, 15 Mar 2019 14:42:03 +0000
Subject: [PATCH 1/2] vendor/getinfo: Iterate through vendors in lexicographic
 order

find(1) works in readdir() order, which isn't guaranteed to be sorted.
If the current vendor is not an exact match for any subdirectory in
vendor/, and is somehow a derivative of more than one of them, we should
at least produce a deterministic answer.

Closes: #924662
Signed-off-by: Simon McVittie <smcv@collabora.com>
---
 vendor/getinfo | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vendor/getinfo b/vendor/getinfo
index 37eb74cef..8830c238f 100755
--- a/vendor/getinfo
+++ b/vendor/getinfo
@@ -6,7 +6,7 @@ BASEDIR="$(readlink -f "$(dirname $0)")"
 getcurrent() {
 	# search for an exact match to use the correct sources.list example
 	cd $BASEDIR
-	DISTROS="$(find . -mindepth 1 -maxdepth 1 -type d | cut -d'/' -f 2)"
+	DISTROS="$(find . -mindepth 1 -maxdepth 1 -type d | cut -d'/' -f 2 | LC_ALL=C sort)"
 	for DISTRO in $DISTROS; do
 		if dpkg-vendor --is $DISTRO; then
 			echo $DISTRO
-- 
2.20.1

